### PR TITLE
B1

### PR DESCRIPTION
**Pull request by YitaoLiu1996**

### CODE REVIEW

File: R/tidy_observation.R [hunk 1]
```diff
@@ -1,3 +1,16 @@
+#' tidy_observation
+#'
+#' Select the Desired Observation
+#'
+#' @param observation_from
```
Comment (elong0527) [inline R/tidy_observation.R:5]: please add explanation of each variable in @param.

File: R/tidy_observation.R [hunk 2]
```diff
@@ -1,3 +1,22 @@
+#' tidy_observation
+#'
+#' Select the Desired Observation
+#'
```
Comment (elong0527) [inline R/tidy_observation.R:4]: This is a good start. We shall make the variable  definition more informative to clarify the type of variable. 

Please refer example below and try to follow similar style. 

Sentence 1: define variable type
Sentence 2: explain what's the meaning of the variable
Sentence 3 (optional): explain what's the default value


```
#' @param population_from A data frame to obtain population level variables.
#'                        Typically an `adsl` dataset.
#'                        It is the source of variables mentioned in
#'                        `population_where`, `treatment_var`, `stratum_var` and `covariate_var`.
#' @param observation_from A data frame to obtain observation level variables.
#'                         Typically an `adtte` dataset for time-to-event analysis.
#'                         It is the source of variables mentioned in
#'                         `observation_where`, `start_date_var`, `end_date_var`, `censor_var`.
#' @param population_where A character string to define the criteria to select analysis population.
#' @param observation_where A character string to define the criteria to select analysis observation.
#' @param stratum_var A character string to define the variable of baseline stratum in `population_from`.
#'        Only one `stratum_var` is allowed.
```


@Yanping123DFEF  @mj719  @kyle9960 , you can follow similar style for your work

File: R/tidy_observation.R [hunk 3]
```diff
@@ -1,3 +1,22 @@
+#' tidy_observation
+#'
+#' Select the Desired Observation
```
Comment (elong0527) [inline R/tidy_observation.R:3]: Please clarify more of what we are trying to achieve. 

What is "Desired Observation"?  You may refer example below

```
#' This function is used to prepare ADaM datasets (e.g. ADSL and ADTTE),
#' for time-to-event analysis. The function filter the population,
#' and then tidy the data as the standard time to event format. So the output
#' data will be easy to fit into the subsequent analysis functions.
```

File: R/tidy_observation.R [hunk 4]
```diff
@@ -1,3 +1,22 @@
...
+#'
+#' @param observation_from The input observation data table. For example, "adverse events information" from the table "adae".
+#' @param observation_where Tells the function which rows of the table should be select, according to the value of certain column. For example, subset(observation_from,observation_where).
+#' @param treatment_var The function will create a new column called "treatment" based on the column of this parameter. For example, the "treatment" column will be equal to "TRTA".
+#' @param treatment_order Tells the function which rows of the table should be select, only if the values in "treatment" is in this parameter. It also provide the label names of the treatments after turning them into factors. For example, include rows which treatment = "Placebo" and treatment = "Xanomeline High Dose", and change the factor labels into "Placebo" and "MK9999".
+#'
+#' @return a table of observation information
```
Comment (elong0527) [inline R/tidy_observation.R:10]: Return shall follow similar style:

what's the type of output?
if it is a data frame, what is the meaning of each variable in the data frame?

**Review by elong0527 — CHANGES_REQUESTED** [review]
It is a good start! I provide some examples for you to refer. They are from other function, you shall not directly copy and paste but just refer them to document yours.

File: R/tidy_observation.R [hunk 5]
```diff
@@ -1,13 +1,76 @@
 #' tidy_observation
 #'
-#' Select the Desired Observation
+#' This function is used to prepare the observation datasets for the use in function tidy_ae_table().
```
Comment (elong0527) [inline R/tidy_observation.R:3]: just add code formatting 
```suggestion
#' This function is used to prepare the observation datasets for the use in function `tidy_ae_table()`.
```

File: R/tidy_observation.R [hunk 6]
```diff
@@ -1,13 +1,76 @@
...
+#' The function filter the 'adae' dataset, and define a 'treatment' in the dataset.
```
Comment (elong0527) [inline R/tidy_observation.R:4]:
```suggestion
#' The function filter the `adae` dataset, and define proper treatment order in the dataset.
```

File: R/tidy_observation.R [hunk 7]
```diff
@@ -1,13 +1,76 @@
...
 #'
-#' @param observation_from The input observation data table. For example, "adverse events information" from the table "adae".
-#' @param observation_where Tells the function which rows of the table should be select, according to the value of certain column. For example, subset(observation_from,observation_where).
-#' @param treatment_var The function will create a new column called "treatment" based on the column of this parameter. For example, the "treatment" column will be equal to "TRTA".
-#' @param treatment_order Tells the function which rows of the table should be select, only if the values in "treatment" is in this parameter. It also provide the label names of the treatments after turning them into factors. For example, include rows which treatment = "Placebo" and treatment = "Xanomeline High Dose", and change the factor labels into "Placebo" and "MK9999".
 #'
-#' @return a table of observation information
+#' @param observation_from A data frame to obtain observation level variables.
+#'                         Typically an 'adae' dataset for time-to-event analysis.
```
Comment (elong0527) [inline R/tidy_observation.R:8]:
```suggestion
#'                         Typically an `adae` dataset for time-to-event analysis.
```

File: R/tidy_observation.R [hunk 8]
```diff
@@ -1,13 +1,76 @@
...
+#'                         It is the source of variables mentioned in
+#'                         'observation_where', 'start_date_var', 'end_date_var', 'censor_var'.
```
Comment (elong0527) [inline R/tidy_observation.R:10]:
```suggestion
#'                         `observation_where`, `start_date_var`, `end_date_var`, `censor_var`.
```

File: R/tidy_population.R [hunk 9]
```diff
@@ -1,15 +1,24 @@
 #' tidy_population
 #'
-#' Get population level tidy data
+#' This function is used to prepare the observation datasets for the use in function tidy_ae_table().
+#' The function filter the 'adsl' dataset, and define a 'treatment' in the dataset.
```
Comment (elong0527) [inline R/tidy_population.R:4]:
```suggestion
#' The function filter the `adsl` dataset, and define proper treatment order in the dataset.
```

File: R/tidy_population.R [hunk 10]
```diff
@@ -1,15 +1,24 @@
...
-#' @param population_where Tells the function which rows of the table should be select, according to the value of certain column. For example, select subset where "ITTFL=='Y'.
-#' @param treatment_var The function will create a new column called "treatment" based on the column of this parameter. For example, the "treatment" column will be equal to "TRT01A".
-#' @param treatment_order Tells the function which rows of the table should be select, only if the values in "treatment" is in this parameter. It also provide the label names of the treatments after turning them into factors. For example, include rows which treatment = "Placebo" and treatment = "Xanomeline High Dose", and change the factor labels into "Placebo" and "MK9999".
-#' @param stratum_var The function will create a new column called "stratum" based on this parameter.
+#' @param population_from A data frame to obtain population level variables.
+#'                        Typically an 'adsl' dataset.
+#'                        It is the source of variables mentioned in
+#'                        'population_where', 'treatment_var', 'stratum_var' and 'covariate_var'.
```
Comment (elong0527) [inline R/tidy_population.R:9]:
```suggestion
#'                        `population_where`, `treatment_var`, `stratum_var` and `covariate_var`.
```

File: R/tidy_population.R [hunk 11]
```diff
@@ -1,15 +1,24 @@
...
+#' @param population_where A character string to define the criteria to select analysis population.
+#' @param treatment_var  A character string to define the variable of new column called "treatment".
+#' @param treatment_order A vector of character strings that tells the function which rows of the table should be select, only if the values in "treatment" is in this vector.
```
Comment (elong0527) [inline R/tidy_population.R:12]:
```suggestion
#' @param treatment_order A vector of character strings that tells the 
#'                function which rows of the table should be select, 
#'                only if the values in "treatment" is in this vector.
```

Comment (elong0527) [reply R/tidy_population.R:12]: Break long line into multiple.

**Review by elong0527 — CHANGES_REQUESTED** [review]
Looks good. If you are OK with my suggestions below. Just accept them and I will approve it to complete this review.

**Comment by YitaoLiu1996** [comment]
Hello Yilong,
I have made the changes according to your suggestions. Please review them.
Also, is there any reading that I could do to get ready for the
next meeting?


yilong zhang ***@***.***> 于2021年3月19日周五 下午9:09写道：

**Comment by elong0527** [comment]
Thanks Yitao. You may review reference in #2 and #3 in preparing the work

**Pull request merged**